core/ide/src/main/java/org/alice/stageide/icons/ShapeIconFactory.java

```diff
@@ -42,28 +42,38 @@
  *******************************************************************************/
 package org.alice.stageide.icons;
 
-import org.lgna.croquet.icon.ResolutionIndependantIconFactory;
+import edu.cmu.cs.dennisc.java.util.logging.Logger;
+import org.lgna.croquet.icon.ResolutionIndependentIconFactory;
 
-import javax.swing.Icon;
-import java.awt.Dimension;
+import java.awt.*;
+import java.lang.reflect.Constructor;
 
 /**
- * @author Dennis Cosgrove
+ * @author Jen Smith
  */
-public class JointIconFactory extends ResolutionIndependantIconFactory {
-  private static class SingletonHolder {
-    private static JointIconFactory instance = new JointIconFactory();
-  }
 
-  public static JointIconFactory getInstance() {
-    return SingletonHolder.instance;
-  }
+public class ShapeIconFactory extends ResolutionIndependentIconFactory {
+  Constructor<? extends ShapeIcon> iconConstructor;
 
-  private JointIconFactory() {
+  public ShapeIconFactory(Class<? extends ShapeIcon> cls)  {
```

Comment:
I was starting to suggest replacing the Class with a direct reference to the new method, but once I started down that path I saw how much more of the icon factory hierarchy can be collapsed or eliminated.
I think I will offer it as possible next step PR.

To be overly precise, the type would become `Function<Dimension, ShapeIcon>`, called with things like `DiscIcon::new` and invoked with `iconConstructor.apply(size)`.

Reply:
I support all ideas to dismantle the icon factory industrial complex!